dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -168,13 +166,16 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataWriter::writeTempPart(BlockWithDa
 	std::shared_ptr<PartLog> part_log = context.getPartLog();
 	if (part_log)
 	{
+		PartLogElement elem;
+		elem.event_time = time(0);
+
 		elem.event_type = PartLogElement::NEW_PART;
 		elem.size_in_bytes = new_data_part->size_in_bytes;
 		elem.duration_ms = stopwatch.elapsed() / 1000000;
 
 		elem.database_name = new_data_part->storage.getDatabaseName();
 		elem.table_name = new_data_part->storage.getTableName();
-		elem.part_name = new_data_part->name;
+		elem.part_name = new_data_part->name.substr(4);
```

Comment:
Very unobvious.
`strlen("tmp_")` should be better, but also not good enough.
Keeping separate variables for temporary name and result name will be better.